packages/citizen-sdk/src/sdks/viem-claim-sdk.ts

```diff
@@ -0,0 +1,368 @@
+      throw new Error(
+        `ClaimSDK: Contract addresses not found for configured env: '${env}'`,
+      )
+    }
+
+    this.ubiSchemeAddress = contractEnvAddresses.ubiContract as Address
+    this.ubiSchemeAltAddress = contractAddresses[this.altChain][env]
+      .ubiContract as Address
+    this.faucetAddress = contractEnvAddresses.faucetContract as Address
+  }
+
+  static async init(
+    props: Omit<ClaimSDKOptions, "account">,
+  ): Promise<ClaimSDK> {
+    const [account] = await props.walletClient.getAddresses()
+    return new ClaimSDK({ account, ...props })
+  }
+
+  /**
+   * Reads a contract function using publicClient.
+   * @param params - Parameters for the contract read operation.
+   * @returns The result of the contract read.
+   * @throws If the contract read fails.
+   */
+  private async readContract<T>(
+    params: {
+      address: Address
+      abi: any
+      functionName: string
+      args?: any[]
+    },
+    altClient?: PublicClient,
+  ): Promise<T> {
+    try {
+      const client = altClient || this.publicClient
+      return (await client.readContract({
+        address: params.address,
+        abi: params.abi,
+        functionName: params.functionName,
+        args: params.args || [],
+        account: this.account,
+      })) as T
+    } catch (error: any) {
+      throw new Error(
+        `Failed to read contract ${params.functionName}: ${error.message}`,
+      )
+    }
+  }
+
+  /**
+   * Submits a transaction and waits for its receipt.
+   * @param params - Parameters for simulating the contract call.
+   * @param onHash - Optional callback to receive the transaction hash.
+   * @returns The transaction receipt.
+   * @throws If submission fails or no active wallet address is found.
+   */
+  async submitAndWait(
+    params: SimulateContractParameters,
+    onHash?: (hash: `0x${string}`) => void,
+  ): Promise<TransactionReceipt> {
+    if (!this.account) {
+      throw new Error("No active wallet address found.")
+    }
+
+    const { request } = await this.publicClient.simulateContract({
+      account: this.account,
+      ...params,
+    })
+
+    const hash = await this.walletClient.writeContract(request)
+    onHash?.(hash)
+
+    // we wait one block
+    // to prevent waitFor... to immediately throw an error
+    await new Promise((res) => setTimeout(res, 5000))
+
+    return waitForTransactionReceipt(this.publicClient, {
+      hash,
+      retryDelay: 5000,
+    })
+  }
+
+  /**
+   * Checks if the connected user is eligible to claim UBI for the current period.
+   * Returns the amount they can claim (0 if not eligible or already claimed).
+   * Does not check for whitelisting status.
+   * @param pClient - Optional public client to check entitlement on alternative chain.
+   * @returns The claimable amount in the smallest unit (e.g., wei).
+   * @throws If the entitlement check fails.
+   */
+  async checkEntitlement(pClient?: PublicClient): Promise<bigint> {
```

Comment:
pclient is in the class already

Reply:
Thats the client for the current connected network.
for the ClaimButton we check if you have an claim available on the other chain using a temporary read-only public client for the opposing chain. code here: https://github.com/GoodDollar/GoodSDKs/blob/b3e2f718277bfe7dd87d064304a41f4cac308ffe/packages/ui-components/src/ClaimButton.ts#L317